tests/ArmTemplates.Tests/Extractor/Scenarios/ApiExtractorByVersionSetNameTests.cs

```diff
@@ -0,0 +1,249 @@
+            mockedApimServiceExtractor
+                .Setup(x => x.GenerateApiManagementServicesTemplateAsync(It.IsAny<ExtractorParameters>()))
+                .ReturnsAsync(new Template<ApiManagementServiceResources>());
+
+            var mockedParametersExtractor = new Mock<IParametersExtractor>(MockBehavior.Strict);
+            mockedParametersExtractor
+                .Setup(x => x.CreateMasterTemplateParameterValues(
+                    It.IsAny<List<string>>(),
+                    It.IsAny<LoggersCache>(),
+                    It.IsAny<LoggerTemplateResources>(),
+                    It.IsAny<BackendTemplateResources>(),
+                    It.IsAny<NamedValuesResources>(),
+                    It.IsAny<IdentityProviderResources>(),
+                    It.IsAny<OpenIdConnectProviderResources>(),
+                    It.IsAny<ExtractorParameters>()))
+                .ReturnsAsync(new Template());
+
+            mockedParametersExtractor
+                .Setup(x => x.CreateResourceTemplateParameterTemplate(It.IsAny<Template>(), It.IsAny<Template>()))
+                .Returns(new Template());
+
+            var extractorExecutor = ExtractorExecutor.BuildExtractorExecutor(
+                this.GetTestLogger<ExtractorExecutor>(),
+                apisClient: mockedApisClient,
+                apiExtractor: mockapiExtractor.Object,
+                policyExtractor: mockedPolicyExtractor.Object,
+                productApisExtractor: mockedProductApisExtractor.Object,
+                productExtractor: mockedProductExtractor.Object,
+                apiVersionSetExtractor: mockedApiVersionSetExtractor.Object,
+                authorizationServerExtractor: mockedAuthorizationServerExtractor.Object,
+                tagExtractor: mockedTagsExtractor.Object,
+                tagApiExtractor: mockedApiTagsExtractor.Object,
+                loggerExtractor: mockedLoggerExtractor.Object,
+                namedValuesExtractor: mockedNamedValuesExtractor.Object,
+                backendExtractor: mockedBackendExtractor.Object,
+                groupExtractor: mockedGroupExtractor.Object,
+                openIdConnectProviderExtractor: mockedOpenIdConnectProvidersExtractor.Object,
+                schemaExtractor: mockedSchemasExtractor.Object,
+                policyFragmentsExtractor: mockedPolicyFragmentExtractor.Object,
+                apiReleaseExtractor: mockedApiReleasesExtractor.Object,
+                parametersExtractor: mockedParametersExtractor.Object,
+                apiManagementServiceExtractor: mockedApimServiceExtractor.Object
+                );
+            extractorExecutor.SetExtractorParameters(extractorParameters);
+
+            // act
+            await extractorExecutor.GenerateAPIVersionSetTemplates();
+
+            // assert
+            var directoriesCreated = Directory.GetDirectories(currentTestDirectory);
```

Comment:
we definitely need to think about separate class which knows how to create directories... (but in next PRs)
And huge thanks for mocking all of the services here :)

Reply:
Always welcome! 
May I clarify what do you mean by creating directories? to which line you are referring ? (this line is getting list of directories)

Reply:
I am just referring to that method inside of `ExtractorExecutor` - if we would have another component, that is responsible for creating directories, we could avoid having that big amount of mocks, but we could explicitly test that component on creation of directories

Reply:
Oh I have created those mocks for future plans to increase coverage by tests :) 
Folders are created as a result of running full cycle of extractors

Reply:
Sure, I am just proposing ideas for future, if we have time.
Resolving comment since there is no suggestion for current PR